site/blog/ecs-simplified.mdx

```diff
@@ -0,0 +1,125 @@
+1. Deploy the folder `examples/ecs-fargate/prisma/app` of the repo `https://github.com/iasql/iasql-engine/` to ECS on my AWS account
+2. Create a security group named `allow-all` attached to the VPC with ID `vpc-f56cff91`. Then add a rule to it allowing all incoming traffic on port `80` from `0.0.0.0/0`.
+
+The first statement, is described in a high-level abstraction layer. The higher the level, the less detail. So it doesn't say anything about the underlying steps that should be taken in order for the statement to be executed. But the second statement is giving all the details necessary to create a new security group.
+
+The flexibility of IaSQL– which makes it really awesome– is that you can create both the high-level and the low-level modules for it. Both of the above statements are already possible using IaSQL modules. The first one using a module named `aws_ecs_simplified` and the latter using the `aws_security_group` module. Let's take a peek on the actual implementations really quick before we continue our main topic.
+
+```sql title="Deploy the folder 'examples/ecs-fargate/prisma/app' of the 'iasql-engine' repo to ECS on my AWS account"
+-- Install the "high-level" aws_ecs_simplified IaSQL module
+SELECT iasql_install('aws_ecs_simplified');
+-- Create the load balancer, ECS service, task definition, etc. All ready afterwards for us to build and push to deploy the app.
+INSERT INTO ecs_simplified (app_name, app_port, image_tag, public_ip) VALUES ('simple-express', 8088, 'latest', true);
+SELECT iasql_apply();
+-- Use the "high-level" ecr_build function to build the folder "examples/ecs-fargate/prisma/app" from "iasql-engine" Github repo and push it to ECR
+SELECT ecr_build(
+  'https://github.com/iasql/iasql-engine/',
+  (SELECT id FROM repository WHERE repository_name = 'simple-express-repository')::varchar(255),
+  './examples/ecs-fargate/prisma/app',
+  'ghp_XXX', -- replace your github personal access token here
+  'main'
+);
+```
+```sql title="Create a security group named 'allow-all' attached to the VPC with ID 'vpc-f56cff91'. Then add a rule to it allowing all incoming traffic on port '80' from '0.0.0.0/0'"
+-- Install the aws_security_group IaSQL module
+SELECT iasql_install('aws_security_group');
+-- "Create a security group named `allow-all` attached to the VPC with ID `vpc-f56cff91`"
+INSERT INTO security_group (description, group_name, vpc_id)
+VALUES ('allows all incoming traffic on port 80', 'allow-all', (SELECT id FROM vpc WHERE vpc_id = 'vpc-08c0a68025c2983a2'));
+-- "Then add a rule to it allowing all incoming traffic on port `80` from `0.0.0.0/0`."
+INSERT INTO security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+VALUES (false, 'tcp', 80, 80, '0.0.0.0/8', 'allow all to 80', (SELECT id FROM security_group WHERE group_name = 'allow-all'));
+```
```

Comment:
Also, maybe we can move the structure of this section to put the SQL examples closer to the point they're describing. Not a blocker, just a suggestion.

I mean, talk about high-level modules and show the example, then talk about the low-level module and show the example.